### PR TITLE
CENTROL-53 implemented /trade command + validations/error handling

### PR DESCRIPTION
**Pull request by bahti-centrol**
This will allow users to create /trade command and pass in their parameters. 

* slash_command_helper.py
  * Added /trade slash command with all appropriate fields (some have single option since we only support one broker)
  * Added slash command support for hello, help, and j
  * Refactored each slash command to log the user data and call appropriate response in discord_response.py
* discord_response.py
  * Added trade handler and logs all incoming parameters
  * Does validations on the parameters to make sure the user is not trying to make bad requests
  * Added Error handling to make sure most edge cases are taken care of the user is notified (previously bot does not even reply)
  * Refactored other functions to accept slash commands
  * Added better logging for the functions to uid is printed for each request
* centrol_discord.py
  * Removed all regular ! commands [CENTROL-58]
* alerts.py
  * Minor refactoring for better logs

[CENTROL-58]: https://centrol.atlassian.net/browse/CENTROL-58?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

* Moved all notifications to DM's. No need to send reply message id

!cflip migration to /cflip:
* Created 3 subcommands:
  * /cflip help => small tutorial on how to use
  * /cflip ratings **token** => accepts one parameter and displays exchange ratings
  * /cflip alerts **token** **exchange** **rating** => creates a traditional cflip alert
 
image

image

### CODE REVIEW

File: src/centrol/user.py [hunk 1]
```diff
@@ -12,23 +12,28 @@ def __init__(self):
         log.info("Loading centrol user module")
         self.config = CentrolConfig()
 
-    def check_user(self, user: str) -> bool:
-        check_url = self.config.SERVER_URL + f"/bot/check_user?user={user}"
-        resp = get(check_url)
+    def check_user(self, uid: str, meta: dict) -> dict:
```
Comment (bahti-centrol) [inline src/centrol/user.py:15]: `check_user()` and `create_user()` was my temp functions while waiting for PR-84. `get_user_data()` and `create_new_user` are the ones that will actually be used. I will remove this in the next commit/PR.

File: src/centrol_bots/centrol_discord.py [hunk 2]
```diff
@@ -71,139 +71,20 @@ async def handle_messages(self, message):
...
-                return await message.author.send(reply)
-
-        # !sell crypto btc 0.00001
-        if message.content.startswith("!sell crypto mkt"):
-            log.debug(f'User ID={message.author.id} used the "!sell crypto mkt" command. Message={message.content}')
-            async with message.channel.typing():
-                reply = await self.dresponse.trade_crypto_mkt(message, 'sell')
-                return await message.author.send(reply)
-
         if message.content.startswith("!add-token coinbase"):
```
Comment (bahti-centrol) [inline src/centrol_bots/centrol_discord.py:74]: This has been moved to slash command by joseph. His PR comes after this merge

**Comment by bahti-centrol** [comment]
Merging this into dev branch so we can start testing it live

**Comment by mikeCentrol** [comment]
Functionally this looks great. Some comments added in line.

**Pull request merged**